### PR TITLE
ddl: fix sequence default value will cause primary-key occur not-null error

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve?

close bug competition issue [#11](https://github.com/tidb-challenge-program/bug-hunting-issue/issues/11)

### What is changed and how it works?
if the default is expr(sequence expr), return nil directly in function `checkDefaultValue`
otherwise will cause some primary key check problem.

### Check List 

Tests 

- Unit test
- Integration test

### CODE REVIEW

File: ddl/sequence_test.go [hunk 1]
```diff
@@ -255,6 +255,11 @@ func (s *testSequenceSuite) TestSequenceAsDefaultValue(c *C) {
 	s.tk.MustExec("create sequence seq")
 
 	// test the use sequence's nextval as default.
+	s.tk.MustExec("drop table if exists t")
+	s.tk.MustExec("create table t(a int not null default next value for seq key)")
+	s.tk.MustExec("drop table if exists t")
+	s.tk.MustExec("create table t(a int not null default nextval(seq), b int, primary key(a))")
```
Comment (zimulala) [inline ddl/sequence_test.go:261]: Could we add other `DefaultIsExpr` tests?

Comment (AilinKid) [reply ddl/sequence_test.go:261]: here the default nextval(seq) is the case of `col.DefaultIsExpr=true`

Comment (zimulala) [reply ddl/sequence_test.go:261]: Yes, but I mean other `DefaultIsExpr` exclude `nextval(seq)`.

Comment (AilinKid) [reply ddl/sequence_test.go:261]: yes, `DefaultIsExpr` is only used by sequence now

**Pull request merged**